### PR TITLE
Extend the warning about the lack of the necessary CUDA libraries

### PR DESCRIPTION
**Pull request by JanuszL**
- when DALI links dynamically to the CUDA libraries (NPP, nvJPEG, cuFFT)
  the user needs to get a meaningful error message when these
  libraries are not installed with the information on how to install it

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

## Category:
**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
- when DALI links dynamically to the CUDA libraries (NPP, nvJPEG, cuFFT)
  the user needs to get a meaningful error message when these
  libraries are not installed with the information on how to install it

### Affected modules and functionalities:
- backend.py

### Key points relevant for the review:
- NA

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [s] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

**Review by stiepan — APPROVED** [review]

File: dali/python/nvidia/dali/backend.py [hunk 1]
```diff
@@ -70,14 +70,26 @@ def check_cuda_runtime():
...
         if GetCufftVersion() == -1:
             deprecation_warning(
-                "Cannot access cuFFT library. Please check cuda installation and/or "
-                "if an appropriate wheel is installed.")
+                "nvidia-dali-cuda120 no longer ships with CUDA runtime. You need to install "
+                "it separately. Please check cuda installation and/or if an appropriate wheel "
+                "that provides cuFFT is installed. Please check "
+                "https://docs.nvidia.com/cuda/cuda-quick-start-guide/"
+                "index.html#pip-wheels-installation-linux "
+                "for the reference")
```
Comment (szalpal) [inline dali/python/nvidia/dali/backend.py:78]: How about using `""" ... """`? Might be easier to edit later.

File: dali/python/nvidia/dali/backend.py [hunk 2]
```diff
@@ -70,14 +70,26 @@ def check_cuda_runtime():
 
         if GetCufftVersion() == -1:
             deprecation_warning(
-                "Cannot access cuFFT library. Please check cuda installation and/or "
-                "if an appropriate wheel is installed.")
+                "nvidia-dali-cuda120 no longer ships with CUDA runtime. You need to install "
```
Comment (szalpal) [inline dali/python/nvidia/dali/backend.py:73]:
```suggestion
                "nvidia-dali-cuda120 is no longer shipped with CUDA runtime. You need to install "
```

File: dali/python/nvidia/dali/backend.py [hunk 3]
```diff
@@ -70,14 +70,26 @@ def check_cuda_runtime():
...
+                "it separately. Please check cuda installation and/or if an appropriate wheel "
+                "that provides cuFFT is installed. Please check "
```
Comment (szalpal) [inline dali/python/nvidia/dali/backend.py:75]: Maybe this version of the sentence would be easier to understood? Just a suggestion.
```suggestion
                "it separately. cuFFT is typically provided with CUDA Toolkit installation or an appropriate wheel. "
                "Please check "
```

File: dali/python/nvidia/dali/backend.py [hunk 4]
```diff
@@ -70,14 +70,26 @@ def check_cuda_runtime():
...
 
         if GetNppVersion() == -1:
-            deprecation_warning("Cannot access NPP library. Please check cuda installation and/or "
-                                "if an appropriate wheel is installed.")
-
+            deprecation_warning(
+                "nvidia-dali-cuda120 no longer ships with CUDA runtime. You need to install "
+                "it separately. Please check CUDA installation and/or if an appropriate wheel "
+                "that provides NPP is installed. Please check "
+                "https://docs.nvidia.com/cuda/cuda-quick-start-guide/"
```
Comment (szalpal) [inline dali/python/nvidia/dali/backend.py:85]: Similar suggestions as above (`"Please check"` phrase repeats at the beginning of sentences in this error message).

File: dali/python/nvidia/dali/backend.py [hunk 5]
```diff
@@ -70,14 +70,26 @@ def check_cuda_runtime():
...
+                "index.html#pip-wheels-installation-linux "
+                "for the reference")
         if GetNvjpegVersion() == -1:
             deprecation_warning(
-                "Cannot access nvJPEG library. Please check cuda installation and/or "
-                "if an appropriate wheel is installed.")
+                "nvidia-dali-cuda120 no longer ships with CUDA runtime. You need to install "
+                "it separately. Please check CUDA installation and/or if an appropriate wheel "
+                "that provides nvJPEG is installed. Please check "
+                "https://docs.nvidia.com/cuda/cuda-quick-start-guide/"
```
Comment (szalpal) [inline dali/python/nvidia/dali/backend.py:93]: Similar suggestions as above.

**Review by szalpal — APPROVED** [review]

File: dali/python/nvidia/dali/backend.py (hunk 1 above)
Comment (JanuszL) [reply dali/python/nvidia/dali/backend.py]: `"""..."""` adds line breaks. I'm not sure if that is what we want.

File: dali/python/nvidia/dali/backend.py (hunk 3 above)
Comment (JanuszL) [reply dali/python/nvidia/dali/backend.py]: Done

File: dali/python/nvidia/dali/backend.py (hunk 2 above)
Comment (JanuszL) [reply dali/python/nvidia/dali/backend.py]: Done

File: dali/python/nvidia/dali/backend.py (hunk 4 above)
Comment (JanuszL) [reply dali/python/nvidia/dali/backend.py]: Done

File: dali/python/nvidia/dali/backend.py (hunk 5 above)
Comment (JanuszL) [reply dali/python/nvidia/dali/backend.py]: Done

**Comment by JanuszL** [comment]
!build

**Pull request merged**